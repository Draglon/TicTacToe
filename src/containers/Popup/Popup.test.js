import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import ConnectedPopup, { Popup } from './index';

describe('<Popup /> container', () => {
  const mockStore = configureStore();
  const initialState = {
    reducers: {
      ui: {
        popupToggle: {
          title: '',
          message: '',
          btnClose: true,
          show: false,
        },
      },
    },
  };
  let store;
  let container;

  beforeEach(() => {
    store = mockStore(initialState);
    container = shallow(<ConnectedPopup store={store} />);
  });

  it('render the connected(SMART) component', () => {
    expect(container.length).toEqual(1);
  });

  // it('check Prop matches with initialState', () => {
  //   expect(container.prop('reducers')).toEqual(initialState.reducers);
  // });

  // it('check action on dispatching ', () => {
  //   // let action;
  //   store.dispatch(popupToggle('POPUP_TOGGLE', {
  //     title: '',
  //     message: '',
  //     btnClose: true,
  //     show: false,
  //   }));
  // });

  it('render property', () => {
    expect(container).toMatchSnapshot();
  });
});
