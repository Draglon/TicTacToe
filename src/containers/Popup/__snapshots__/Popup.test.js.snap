// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popup /> container render property 1`] = `ShallowWrapper {}`;
